packages/java-parser/src/unicode.js

```diff
@@ -0,0 +1,144 @@
+// Below adding the isIdentifierIgnorable characters from Java Specs but not from FORMAT general category
+for (let i = 0; i < 8; i++) {
+  manuallyAddedCharacters.add(i);
+}
+for (let i = parseInt("000E", 16); i < parseInt("001B"); i++) {
+  manuallyAddedCharacters.add(i);
+}
+for (let i = parseInt("007F", 16); i < parseInt("009F"); i++) {
+  manuallyAddedCharacters.add(i);
+}
+
+// Below adding Combining Marks
+for (let i = parseInt("0300", 16); i < parseInt("036F"); i++) {
+  manuallyAddedCharacters.add(i);
+}
+for (let i = parseInt("1AB0", 16); i < parseInt("1AFF"); i++) {
+  manuallyAddedCharacters.add(i);
+}
+for (let i = parseInt("1DC0", 16); i < parseInt("1DFF"); i++) {
+  manuallyAddedCharacters.add(i);
+}
+for (let i = parseInt("20D0", 16); i < parseInt("20FF"); i++) {
+  manuallyAddedCharacters.add(i);
+}
+for (let i = parseInt("FE20", 16); i < parseInt("FE2F"); i++) {
+  manuallyAddedCharacters.add(i);
+}
+
+// Adding digits
+for (let i = 48; i <= 57; i++) {
+  manuallyAddedCharacters.add(i);
+}
+
+// Constructing the whole JavaIdentifierPart category.
+const restIdentCharCategories = new Set(
+  (function*() {
+    yield* firstIdentCharCategories;
+    yield* restIdentUnicodeCategories;
+  })()
+);
+
+// Merging all the accepted characters.
+const categories = new Set(
+  (function*() {
+    yield* firstIdentCharCategories;
+    yield* restIdentCharCategories;
+  })()
+);
+
+// Function that pushes in an object an attribute to store the characters
+function pushInUnicode(a, b) {
+  if (unicode.hasOwnProperty(a) === false) {
+    unicode[a] = [parseInt(b)];
+  } else {
+    unicode[a].push(parseInt(b));
+  }
+}
+
+let oldValue; // Algorithm purpose
+lines.forEach(line => {
+  const theLine = line.split(";");
+  if (!categories.has(theLine[2])) {
+    return;
+  }
+  if (theLine[1].match(/Last>$/)) {
+    for (
+      let i = parseInt(oldValue, 16) + 1;
+      i <= parseInt(theLine[0], 16);
+      i++
+    ) {
+      pushInUnicode(theLine[2], i);
+    }
+  } else {
+    pushInUnicode(theLine[2], parseInt(theLine[0], 16));
+  }
+  oldValue = theLine[0];
+});
+
+// Generating a unicodesets.js file so that we don't have to reparse the file each time the parser is called.
+function generateFile() {
+  let data = `"use strict"
+  const firstIdentChar = new Set([`;
+  firstIdentCharCategories.forEach(el => {
+    unicode[el].forEach(value => {
+      data += `${value},`;
+    });
+  });
+  data += `]);
+  `;
+
+  data += `const restIdentChar = new Set([`;
+  restIdentCharCategories.forEach(el => {
+    unicode[el].forEach(value => {
+      data += `${value},`;
+    });
+  });
+  manuallyAddedCharacters.forEach(v => (data += `${v},`));
+
+  data += `]);
+  `;
+  data += `module.exports = {
+    firstIdentChar,
+    restIdentChar
+  }`;
+  fs.writeFileSync(path.resolve(__dirname, "unicodesets.js"), data, err => {
+    if (err) {
+      throw err;
+    }
+  });
+}
+
+// Calling the generation of the file.
+generateFile();
```

Comment:
its wierd that reading the unicode is "inlined" in the source but writing the output
is wrapped in a function.

It would be cleaner if each step would be wrapped in a function so the end of the file would describe a **flow**:
```javascript

const unicodeData = readUnicodeData();
const {restCharCodes, firstCharCodes} = analyzeData(unicodeData);
generateJSUnicodeFile(restCharCodes, firstCharCodes);
```